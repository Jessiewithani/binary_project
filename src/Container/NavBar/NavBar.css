.nav-bar {
    align-items: center;
    border-bottom: 3px solid #E89242FF;
    background-color: rgb(51, 51, 60);
    display: flex;
    height: 150px;
    justify-content: space-between;
    padding: 0 50px 0 50px;
    
}

.nav-button {
    background-color: rgb(51, 51, 60);
    border: none;
    color: #E89242FF;
    font-size: 2em;
}

.nav-button:after {
    border-bottom: 3px solid gainsboro;
    content: '';
    height: 1em;
    left: 90px;
    margin-top: 10px;
    opacity: 0;
    position: absolute;
    transition: opactity 0.35s, transform 0.35s;
    transform: scale(0.1);
    width: 25%;
}
.nav-button:hover:after {
    opacity: 1;
    transform: scale(1)
}

input {
    border: none;
    border-bottom: 3px solid #E89242FF;
    background-color: rgb(51, 51, 60);
    color: gainsboro;
    font-size: 2.5em;
    padding-left: 10px;
    width: 500px;
    
}

