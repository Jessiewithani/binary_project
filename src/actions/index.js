export const saveCharacters = (characters) => ({
    type: 'SAVE_CHARACTERS',
    characters
})

export const saveFavorites = (favorites) => ({
    type: 'SAVE_FAVORITES',
    favorites
})

export const removeFavorites = (id) => ({
    type: 'REMOVE_FAVORITES',
    id
})

export const searchCharacter = (name) => ({
    type: 'SEARCH_CHARACTERS',
    name
})