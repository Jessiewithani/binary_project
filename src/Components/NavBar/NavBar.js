import React, { Component } from 'react';
import '../NavBar/NavBar.css'
import { connect } from 'react-redux';
import { searchCharacter, filterSpecies } from '../../actions'


class NavBar extends Component {
    constructor() {
        super()
        this.state = {
            search: '',
            human: [],
            alien: []

        }
    }
    handleChange = event => {
        this.setState({search: event.target.value})
        this.props.searchCharacter(event.target.value)
        
    }
    
    handleSubmit = (event) => {
        event.preventDefault();
        const { searchCharacter } = this.props;
        searchCharacter(this.state.search)
        this.resetInputs();
    }
    
    resetInputs = () => {
        this.setState({search: ''})
    }

    submitHumanSpecies = event => {
        event.preventDefault();
        const { filterSpecies } = this.props;
        const human = this.props.characters[0].results.filter(char => char.species === 'Human')
        
        // const alien = this.props.characters[0].results.filter(char => char === 'Alien')
        this.setState({ human })
        
        filterSpecies(human)
        // filterSpecies(alien)
        
    }
    
    render() {
        return(
            <div className="nav-bar">
                <div>
                <button>ALL CHARACTERS</button>
                <button>ALIENS</button>
                <button onClick={this.submitHumanSpecies}>HUMANS</button>
                </div>
                <h2>HI NAV</h2>
                <div>
                <input type="text" placeholder="search" name="search" value={this.state.search} onChange={this.handleChange}/>
                <button  onClick={this.handleSubmit}>SEARCH</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    characters: state.charactersReducer,
})

const mapDispatchToProps = (dispatch) => ({
    searchCharacter: name => dispatch(searchCharacter(name)),
    filterSpecies: species => dispatch(filterSpecies(species))
})


export default connect(mapStateToProps, mapDispatchToProps)(NavBar)