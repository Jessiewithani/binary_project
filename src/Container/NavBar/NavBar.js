import React, { Component } from 'react';
import '../NavBar/NavBar.css'
import { connect } from 'react-redux';
import { searchCharacter, filterSpecies } from '../../actions'


export class NavBar extends Component {
    constructor() {
        super()
        this.state = {
            search: '',
            human: [],
            alien: []

        }
    }
    handleChange = event => {
        this.setState({search: event.target.value})
        this.props.searchCharacter(event.target.value)
        
    }
    
    handleSubmit = (event) => {
        event.preventDefault();
        const { searchCharacter } = this.props;
        searchCharacter(this.state.search)
        this.resetInputs();
    }
    
    resetInputs = () => {
        this.setState({search: ''})
    }

    submitHumanSpecies = event => {
        event.preventDefault();
        const { filterSpecies } = this.props;

        // console.log('FILTERSPECIES', filterSpecies)
        const human = this.props.characters[0].results.filter(char => char.species === 'Human')
        console.log('MAPCHARACTERS', this.props.characters)
        this.setState({ human })
        
        filterSpecies(human)
        
    }
    
    render() {
        return(
            <div className="nav-bar">
                <button className="nav-button" onClick={this.submitHumanSpecies}>SHOW  HUMANS  ONLY</button>
                <input type="text" placeholder="search characters" name="search" value={this.state.search} onChange={this.handleChange}/>
            </div>
        )
    }
}

export const mapStateToProps = (state) => ({
    characters: state.charactersReducer,
})

export const mapDispatchToProps = (dispatch) => ({
    searchCharacter: name => dispatch(searchCharacter(name)),
    filterSpecies: species => dispatch(filterSpecies(species))
})


export default connect(mapStateToProps, mapDispatchToProps)(NavBar)