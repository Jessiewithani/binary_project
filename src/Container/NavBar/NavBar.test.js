import React from 'react';
import { NavBar, mapDispatchToProps } from '../NavBar/NavBar';
import { searchCharacter, filterSpecies } from '../../actions';
import { shallow } from 'enzyme'

describe('NavBar', () => {
    it('should match the snapshot with the correct data passed in', () => {
      const wrapper = shallow(<NavBar/>)
      expect(wrapper).toMatchSnapshot();
    })
  })

describe('mapDispatchToProps in navbar', () => {
  it('calls dispath with a searchCharacter action when searchCharacter is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = searchCharacter(17, 'Annie', 'Alive', 'Human', '', 'Female', 'Anatomy Park', 'https://rickandmortyapi.com/api/character/avatar/17.jpeg')

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.searchCharacter(17, 'Annie', 'Alive', 'Human', '', 'Female', 'Anatomy Park', 'https://rickandmortyapi.com/api/character/avatar/17.jpeg')

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
  it('calls dispatch with a filterSpecies action when filterSpecies is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = filterSpecies(17, 'Annie', 'Alive', 'Human', '', 'Female', 'Anatomy Park', 'https://rickandmortyapi.com/api/character/avatar/17.jpeg');

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.filterSpecies(17, 'Annie', 'Alive', 'Human', '', 'Female', 'Anatomy Park', 'https://rickandmortyapi.com/api/character/avatar/17.jpeg');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
  