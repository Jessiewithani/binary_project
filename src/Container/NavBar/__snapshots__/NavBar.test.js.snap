// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot with the correct data passed in 1`] = `ShallowWrapper {}`;
